src/components/Gradebook/index.jsx

```diff
@@ -13,57 +13,64 @@ import {
 import queryString from 'query-string';
 import { configuration } from '../../config';
 import PageButtons from '../PageButtons';
+import { formatDateForDisplay } from '../../data/actions/utils';
 
 const DECIMAL_PRECISION = 2;
+const GRADE_OVERRIDE_HISTORY_COLUMNS = [{ label: 'Date', key: 'date' }, { label: 'Grader', key: 'grader' },
+  { label: 'Reason', key: 'reason' },
+  { label: 'Adjusted grade', key: 'adjustedGrade' }];
 
 export default class Gradebook extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       filterValue: '',
       modalOpen: false,
-      modalModel: [{}],
-      updateVal: 0,
+      adjustedGradeValue: 0,
       updateModuleId: null,
       updateUserId: null,
+      reasonForChange: '',
     };
     this.fileFormRef = React.createRef();
     this.fileInputRef = React.createRef();
+    this.myRef = React.createRef();
   }
 
   componentDidMount() {
     const urlQuery = queryString.parse(this.props.location.search);
     this.props.getRoles(this.props.courseId, urlQuery);
+    this.overrideReasonInput.focus();
   }
 
-  setNewModalState = (userEntry, subsection) => {
-    let adjustedGradePossible = '';
-    let currentGradePossible = '';
-    if (subsection.attempted) {
-      adjustedGradePossible = ` / ${subsection.score_possible}`;
-      currentGradePossible = `/${subsection.score_possible}`;
-    }
-    this.setState({
-      modalModel: [{
-        username: userEntry.username,
-        currentGrade: `${subsection.score_earned}${currentGradePossible}`,
-        adjustedGrade: (
-          <span>
-            <input
-              style={{ width: '25px' }}
-              type="text"
-              onChange={event => this.setState({ updateVal: event.target.value })}
-            />{adjustedGradePossible}
-          </span>
-        ),
-        assignmentName: `${subsection.subsection_name}`,
-      }],
-      modalOpen: true,
-      updateModuleId: subsection.module_id,
-      updateUserId: userEntry.user_id,
-    });
+  onChange(e) {
+    this.setState({ [e.target.name]: e.target.value });
```

Comment:
Nit: I would prefer a more explicit approach to writing these handlers, i.e. two separate & specifically named methods which handle only their specific use cases and can't be repurposed into something else.

I generally prefer to be very concrete in this sort of context. This method doesn't encapsulate enough to be worth abstracting.